sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/InMemoryTableCatalog.scala

```diff
@@ -84,28 +85,28 @@ class BasicInMemoryTableCatalog extends TableCatalog {
     invalidatedTables.add(ident)
   }
 
-  // TODO: remove it when no tests calling this deprecated method.
+  // TODO: remove it when the deprecated method `createTable(..., StructType, ...)`
```

Comment:
Should we remove this TODO? Isn't the current solution the final one?

Reply:
I'm actually a bit hesitant, because if we `eventually` decide to `completely remove` this in `a future version`, keeping it now may be a good reminder, and I want to hear everyone's opinions. @cloud-fan

Reply:
I think we can remove the TODO now. We should never remove the API and implementation will have to provide a fake implementation that just fails.

Reply:
I suggest adding some new comments to explain the reason for retaining this API and throwing exceptions by default while removing the `TODO`

Reply:
> I think we can remove the TODO now. We should never remove the API and implementation will have to provide a fake implementation that just fails.

Done.